src/components/FileRepo/ui.js

```diff
@@ -114,7 +116,36 @@ export const QuerySharingContainer = styled(Row)`
   background: ${({ theme }) => theme.backgroundGrey};
 `;
 
-export const ControlledIcon = props => <ControlledAccessIcon width={12} height={12} {...props} />;
+
+export const ControlledIcon = withTheme((props) => {
+  const initialState = { mouseIsOver: false };
+  const iconContainerRef = React.createRef();
+
+  const didMount = ({ setState }) => {
+    iconContainerRef.current.addEventListener('mouseenter', e => {
```

Comment:
Would it be better to add `onMouseLeave` and `onMouseEnter` as props to the svg icon rather than adding a `ref` and manually doing event listeners seen as React already has those? @hlminh2000 @nyanofthemoon

Reply:
@ciaranschutte good point actually, I just never used the `onMouseLeave` and `onMouseEnter` callbacks on react components so didn't think of that, but that is definitely preferable :)

also @nyanofthemoon I think I know why the event listeners attached on svg wasn't working for you :)